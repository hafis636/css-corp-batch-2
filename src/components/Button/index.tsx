import React, { ComponentProps, VFC, SVGProps } from 'react';

type Props = {} & ComponentProps<'button'>;

const Button = ({ children, icon: Icon, ...props }: Props) => {
  return (
    <button
      {...props}
      className="group relative w-full flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
    >
      {Icon && (
        <span className="absolute left-0 inset-y-0 flex items-center pl-3">
          <Icon />
        </span>
      )}
      {children}
    </button>
  );
};

export default Button;
