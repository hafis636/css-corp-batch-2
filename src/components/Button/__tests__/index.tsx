import React, { ComponentProps } from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from '../index';
import LockIcon from '@assets/icons/lock.svg';

type Props = {
  icon?: React.ElementType;
} & ComponentProps<'button'>;

const setup = ({ children = 'Sign In', icon, ...props }: Props) => {
  return render(<Button {...props}>{children}</Button>);
};

test('should render Button component', () => {
  const { queryByTestId } = setup({});
  const btn = queryByTestId('btn-component');
  expect(btn).not.toBeNull();
  expect(btn?.innerHTML).toBe('Sign In');
});

test('should button disable if disable prop pass', () => {
  const { queryByRole } = setup({ disabled: true });
  const btn = queryByRole('button');
  expect(btn).toBeDisabled();
  expect(btn).toHaveClass('bg-gray-600');
});

test('should button clickable', () => {
  const mockBtn = jest.fn();

  const { queryByRole } = setup({ onClick: mockBtn });
  const btn = queryByRole('button');
  if (btn) {
    fireEvent.click(btn);
    expect(mockBtn).toBeCalledTimes(1);
  }
});

test('should icon visible', () => {
  const { queryByRole, queryByTestId } = setup({ icon: <LockIcon /> });
  const btn = queryByRole('button');
  const span = queryByTestId('icon-span');
  expect(btn).toContainElement(span);
});
