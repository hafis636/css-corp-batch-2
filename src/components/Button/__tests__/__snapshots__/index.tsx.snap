// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`positive scenarios take snapshot of basic component 1`] = `
<button
  class="group relative w-full flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-400"
  data-testid="btn-component"
>
  Sign In
</button>
`;
